Prepare TwG links to Publisher Center

nfmohit opened the pull request:
## Summary

<!-- Please reference the issue this PR addresses in the following list. -->
Addresses issue:

- #5722 

## Relevant technical choices

This PR prepares the links to TwG Publisher Center.

## PR Author Checklist

- [ ] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [ ] My code is backward-compatible with WordPress 4.7 and PHP 5.6.
- [ ] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [ ] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [ ] Run the code.
- [ ] Ensure the acceptance criteria are satisfied.
- [ ] Reassess the implementation with the IB.
- [ ] Ensure no unrelated changes are included.
- [ ] Ensure CI checks pass.
- [ ] Check Storybook where applicable.
- [ ] Ensure there is a QA Brief.

## Merge Reviewer Checklist

- [ ] Ensure the PR has the correct target branch.
- [ ] Double-check that the PR is okay to be merged.
- [ ] Ensure the corresponding issue has a ZenHub release assigned.
- [ ] Add a changelog message to the issue.

eugene-manuilov commented on assets/js/modules/thank-with-google/datastore/publications.js:
```diff
@@ -155,6 +162,39 @@ const baseSelectors = {
+		( select ) => () => {
+			const homeURL = select( CORE_SITE ).getHomeURL();
+
+			const url = new URL( 'https://publishercenter.google.com/' );
+			url.searchParams.set( 'sk_url', encodeURIComponent( homeURL ) );
```
For consistency reason, we should use the [`addQueryArgs`](https://github.com/WordPress/gutenberg/tree/trunk/packages/url#addqueryargs) function instead of dealing with the `URL` class directly.

nfmohit replied:
Addressed this, thank you!

eugene-manuilov commented on assets/js/modules/thank-with-google/datastore/publications.js:
```diff
@@ -155,6 +162,39 @@ const baseSelectors = {
+	getServicePublicationURL: ( _state, publicationID ) => {
+		invariant( publicationID, 'A publicationID is required.' );
+
+		return `https://publishercenter.google.com/publications/${ publicationID }/overview`;
```
We should probably encode the `publicationID` value with the `encodeURIComponent` function.

nfmohit replied:
Addressed this, thank you!

eugene-manuilov commented on assets/js/modules/thank-with-google/datastore/publications.test.js:
```diff
@@ -348,5 +352,67 @@ describe( 'modules/thank-with-google publications', () => {
+				provideUserInfo( registry );
+
+				const homeURL = registry.select( CORE_SITE ).getHomeURL();
+				const publisherCenterURL = new URL(
+					'https://publishercenter.google.com/'
+				);
+				publisherCenterURL.searchParams.set(
+					'sk_url',
+					encodeURIComponent( homeURL )
+				);
```
Same here, let's stick with `addQueryArgs`.

nfmohit replied:
Addressed this as well, thank you!

eugene-manuilov commented on assets/js/modules/thank-with-google/datastore/publications.test.js:
```diff
@@ -348,5 +352,67 @@ describe( 'modules/thank-with-google publications', () => {
+					.select( MODULES_THANK_WITH_GOOGLE )
+					.getServiceCreatePublicationURL();
+
+				expect( createPublicationURL ).toEqual(
```
```suggestion
				expect( createPublicationURL ).toBe(
```

nfmohit replied:
Addressed this, thank you!

eugene-manuilov commented on assets/js/modules/thank-with-google/datastore/publications.test.js:
```diff
@@ -348,5 +352,67 @@ describe( 'modules/thank-with-google publications', () => {
+			} );
+
+			it( 'returns a publisher center URL for an existing publication', () => {
+				registry
+					.dispatch( MODULES_THANK_WITH_GOOGLE )
+					.receiveGetPublications(
+						publicationWithOnboardingCompleteState
+					);
+
+				registry
+					.dispatch( MODULES_THANK_WITH_GOOGLE )
+					.setPublicationID( 'test-publication-a' );
+
+				const publication = registry
+					.select( MODULES_THANK_WITH_GOOGLE )
+					.getCurrentPublication();
```
Why do we need it here? Seems like we don't use neither the current publication nor publications in the `getServicePublicationURL` selector, right?

nfmohit replied:
You're right. I think I was overthinking this one. I've removed them and simplified it.

eugene-manuilov replied:
No worries, all good 👍

eugene-manuilov replied:
Oups wrong comment 😅

eugene-manuilov reviewed (changes requested):
Thanks, @nfmohit. Good work. Added a few comments, please, take a look.

nfmohit commented:
> Thanks, @nfmohit. Good work. Added a few comments, please, take a look.

Thank you for the kind review, @eugene-manuilov! I've addressed the changes.

eugene-manuilov commented on assets/js/modules/thank-with-google/datastore/publications.js:
```diff
@@ -16,6 +16,12 @@
  * limitations under the License.
  */
 
+/**
+ * External dependencies
+ */
+import invariant from 'invariant';
+import { addQueryArgs } from '@wordpress/url';
```
```suggestion
/**
 * External dependencies
 */
import invariant from 'invariant';

/**
 * WordPress dependencies
 */
import { addQueryArgs } from '@wordpress/url';
```

nfmohit replied:
Uh oh! I'm so sorry for overlooking that and thank you for noticing!

eugene-manuilov replied:
No worries, all good 👍

eugene-manuilov reviewed (changes requested):
Thanks, @nfmohit. One minor suggestion that I'll approve by myself and good to go.

felixarntz commented on assets/js/modules/thank-with-google/datastore/publications.js:
```diff
@@ -155,6 +167,42 @@ const baseSelectors = {
+			const homeURL = select( CORE_SITE ).getHomeURL();
+
+			const url = addQueryArgs( 'https://publishercenter.google.com/', {
+				sk_url: encodeURIComponent( homeURL ),
```
This `encodeURIComponent` call needs to be removed, since `addQueryArgs` already handles encoding of parameters. Therefore now the parameter is double encoded which is incorrect.
